allen-cell-animated/timelapse-colorizer
# feature: Export annotations as a CSV (annotations pt. 5)

Problem
=======
Part 5 of 5(?) for #484, annotation prototype.

This change adds the ability to export a CSV containing the annotation data! Each object ID is a row and the labels are listed as columns. 

*Estimated review size: small, 15 minutes (faster if whitespace changes are disabled)*

Solution
========
- Adds a CSV export button in the annotation.
- Adds a new `toCsv()` method in `AnnotationData`, which serializes the annotations to a CSV string.
- Added unit tests for handling bad label names and inputs.

## Type of change

* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open preview link: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview-internal/pr-504/
2. Enable annotation mode and annotate several cells with different labels.
3. Click the "Export as CSV" button to save a CSV file.
4. Open the CSV file. It should render with the correct header names in Excel.

Screenshots (optional):
-----------------------

**Video Talk-through (🔊):**

https://github.com/user-attachments/assets/d2193151-89de-4d2e-99d5-abc4e4d8e03a

![image](https://github.com/user-attachments/assets/3ea3763c-f38c-4602-aeec-673d81f7e34a)
